lib/inferno/dsl/runnable.rb

```diff
@@ -464,6 +464,32 @@ def required_suite_options(suite_option_requirements)
           end
       end
 
+      # Replace a child test/group
+      #
+      # @param id_to_replace [Symbol, String] The ID of the child to be replaced.
+      # @param replacement_id [Symbol, String] The ID of the child that will take the
+      #   place of the child being replaced.
+      # @yield [Inferno::TestGroup, Inferno::Test] Optional block executed in the
+      #    context of the replacement child for additional configuration.
+      # @example
+      #   replace :test2, :test4 do
+      #     id :new_test_id
+      #     config(...)
+      #   end
+      #
+      def replace(id_to_replace, replacement_id, &)
+        index = children.find_index { |child| child.id.to_s.end_with? id_to_replace.to_s }
+        return unless index
+
+        replacement_child = children.find { |child| child.id.to_s.end_with? replacement_id.to_s }
+        return unless replacement_child
+
+        remove(replacement_id)
+        replacement_child.instance_eval(&) if block_given?
+
+        children[index] = replacement_child
```

Comment:
This is going to need to use `group from:` or `test from:` to do all of the extra things that happen when adding a runnable. So I think you'll need to do one of those to add it, then put it in the right place.

Reply:
is there a way to determine if a global id is for a test or a group?

Reply:
Not beyond looking in both repos and seeing where you find it. It would probably be safe to look at the type of the runnable being removed and use that.